src/daft-core/src/array/ops/cast.rs

```diff
@@ -1408,6 +1409,87 @@ impl TensorArray {
                 );
                 Ok(tensor_array.into_series())
             }
+            DataType::SparseTensor(inner_dtype) => {
+                let shape_iterator = self.shape_array().into_iter();
+                let data_iterator = self.data_array().into_iter();
+                let validity = self.data_array().validity();
+                let shape_and_data_iter = shape_iterator.zip(data_iterator);
+                let zero_series = Int64Array::from((
+                    "item",
+                    Box::new(arrow2::array::Int64Array::from_iter([Some(0)].iter())),
+                ))
+                .into_series();
+                let mut non_zero_values = Vec::new();
+                let mut non_zero_indices = Vec::new();
+                let mut offsets = Vec::<usize>::new();
+                for (i, (shape_series, data_series)) in shape_and_data_iter.enumerate() {
+                    let is_valid = validity.map_or(true, |v| v.get_bit(i));
+                    if !is_valid {
+                        // Handle invalid row by populating dummy data.
+                        offsets.push(1);
```

Comment:
It doesn't seem like these offsets are used. What we would do here is push the `count_so_far` into this offset vec that can be converted directly into `Offsets<i64>`

Reply:
Correct the offsets and dummy values doesnt contribute anything to the final values in the series. I kept them in order to preserve the validity of the original dense tensor.